pkgs/by-name/hy/hypercore/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  writeScript,
+}:
+
+buildNpmPackage (finalAttrs: {
+  pname = "hypercore";
+  version = "11.6.2";
+
+  src = fetchFromGitHub {
+    owner = "holepunchto";
+    repo = "hypercore";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-o4Ojja+8Yvn1fK9FszcH/wHZnuyhpHxsch0D1kb5HlE=";
+  };
+
+  npmDepsHash = "sha256-lBV5IehqP2qOe3/mJjLJt8Ty77QN8uUGc5oolFUEqck=";
+
+  dontNpmBuild = true;
+
+  passthru.updateScript = writeScript "update-hypercore" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell -i bash -p  common-updater-scripts curl
+
+    set -eu -o pipefail
+
+    version="$(curl -s https://api.github.com/repos/holepunchto/hypercore/tags | jq -r '.[0].name' | sed 's/^v//')"
+
+    update-source-version hypercore "$version"
+  '';
+
+  meta = {
+    description = "Hypercore is a secure, distributed append-only log.";
+    homepage = "https://github.com/holepunchto/hypercore.git";
+    license = lib.licenses.gpl3Plus;
+    teams = with lib.teams; [ ngi ];
+    maintainers = [ lib.maintainers.goodylove ];
+    platforms = lib.platforms.all;
+  };
+
+  postPatch = ''
+    cp ${./package-lock.json} ./package-lock.json
+  '';
+
```

Comment:
I think it would be more readable if this were to be moved above `passthru.updateScript`